dubbo-metrics/dubbo-metrics-api/src/main/java/org/apache/dubbo/metrics/event/EventBus.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dubbo.metrics.event;
+
+import org.apache.dubbo.common.beans.factory.ScopeBeanFactory;
+import org.apache.dubbo.rpc.model.ApplicationModel;
+
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * Dispatches events to listeners, and provides ways for listeners to register themselves.
+ */
+public class EventBus {
+
+    /**
+     * Posts an event to all registered subscribers and only once.
+     *
+     * @param event event to post.
+     */
+    public static void solo(MetricsEvent event) {
+        if (event.getSource() instanceof ApplicationModel) {
```

Comment:
It is possible that source here is not a ApplicationModel? If so, we should change these usage. Otherwise, change the type of source

Reply:
Modify it to only be used as ApplicationModel